linkerd/proxy/detect/src/lib.rs

```diff
@@ -128,15 +125,12 @@ where
                             target.take().expect("polled after complete"),
                             io.prefix().as_ref(),
                         );
-                        let inner_fut = accept.accept((target, BoxedIo::new(io)));
```

Comment:
Just noting that there was a bug lurking in this change: we were no longer polling the accept to ready before dispatching. These changes were just superficial anyway, it seems, so I've backed this out to be as it was before.